pkg/service/service.go

```diff
@@ -155,6 +158,27 @@ func NewService(monitorNotify monitorNotify) *Service {
 		healthServer:    localHealthServer,
 		lbmap:           lbmap.New(),
 	}
+	// The service update time lastUpdatedTs is initialized here.
+	// If we don't initialize lastUpdatedTs here, then calling the
+	// /healthz endpoint will make cilium-agent to panic since it
+	// would then return an invalid nil interface.
+	svc.lastUpdatedTs.Store(time.Now())
+	return svc
+}
+
+func (s *Service) GetLastUpdatedTs() time.Time {
+	var ts time.Time
+	if val := s.lastUpdatedTs.Load(); val != nil {
+		ts, ok := val.(time.Time)
```

Comment:
This is creating a new `ts` variable. What is the motivation for `var ts time.Time` at the line 170?

Reply:
Let me check that out.

Reply:
> This is creating a new `ts` variable. What is the motivation for `var ts time.Time` at the line 170?

The initial motivation of the declaring the 'var ts time.Time' at line 170 was to return a default value if the 'if statement at line 171' fails.  Please let me know if we should throw an error if we fail at line 171 instead of returning a 'ts' variable.

Reply:
@soumynathan what Martynas was saying is that there's no point on having the variable declared on line 170 since we are creating a new variable on line 172.

We should remove line 170 and `return time.Time` on line 177

Reply:
Ok I will push in a new version.

Reply:
> @soumynathan what Martynas was saying is that there's no point on having the variable declared on line 170 since we are creating a new variable on line 172.
> 
> We should remove line 170 and `return time.Time` on line 177

I don't think we can return just 'time.Time' at line 177. It throws an error 'type time.Time is not an expression'. So I have made a minor tweak to declare the variable before line 172 to have the same effect.

Reply:
Addressed review comments.

Reply:
Curious: what does time.Time{} defaults to? zero or now()? 

In line 165 we initialize it to now(), so when we fallback in line 176, would be good if we stay consistent and also return now().

Reply:
> Curious: what does time.Time{} defaults to? zero or now()?

It defaults to `time.Time`s zero value, which is January 1, year 1, 00:00:00.000000000 UTC according to https://golang.org/pkg/time/#Time.

> In line 165 we initialize it to now(), so when we fallback in line 176, would be good if we stay consistent and also return now().

:+1: agree, initializing it to `now()` is better.

Reply:
> > Curious: what does time.Time{} defaults to? zero or now()?
> 
> It defaults to `time.Time`s zero value, which is January 1, year 1, 00:00:00.000000000 UTC according to https://golang.org/pkg/time/#Time.
> 
> > In line 165 we initialize it to now(), so when we fallback in line 176, would be good if we stay consistent and also return now().
> 
>  agree, initializing it to `time.Now()` is better, also see [#11733 (comment)](https://github.com/cilium/cilium/pull/11733#issuecomment-687220562)

If we all agree for setting the default to 'time.Now()' instead of 'zero value', then I can push a new version.

Reply:
Addressed review comments.